### PR TITLE
feat(examples): Add multiple netif demo: eth+wifi+PPP

### PR DESCRIPTION
**Pull request by david-cermak**
Example running three interfaces, all clients connecting to an endpoint on the internet.
It demonstrates using default netifs, DNS settings per interfaces, prio-routing config option and manual override of default interface.

### CODE REVIEW

**Comment by euripedesrocha** [comment]
I missed the presence of a short README explaining the example and it's goals. Also, the error handling in the `netif` creation, but I think that adding it here could have polluted the example.

File: examples/esp_netif/multiple_netifs/main/multi_netif.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        }
+
+        esp_err_t connect_status = check_connectivity(HOST);
+        if (connect_status == ESP_OK) {
+            // connectivity ok
+            continue;
+        }
+        if (connect_status == ESP_ERR_NOT_FOUND) {
+            // set the default DNS info to global DNS server list
+            // (should be performed automatically in IDF)
```
[the remark this answers is not in the document]
Comment (david-cermak) [reply examples/esp_netif/multiple_netifs/main/multi_netif.c]: This was just my reminder to fix this somehow in IDF. I'll do it directly in `esp_netif`, but for now I have just removed the comment.

**Comment by david-cermak** [comment]
> I missed the presence of a short README explaining the example and it's goals. Also, the error handling in the `netif` creation, but I think that adding it here could have polluted the example.

Thanks for taking a look. Have added the `README.md` with some description.

File: examples/esp_netif/multiple_netifs/main/ppp_connect.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        ppp_info->parent.connected = false;
+    } else if (event_id == IP_EVENT_GOT_IP6) {
+        ESP_LOGI(TAG, "GOT IPv6 event!");
+        ip_event_got_ip6_t *event = (ip_event_got_ip6_t *)event_data;
+        ESP_LOGI(TAG, "Got IPv6 address " IPV6STR, IPV62STR(event->ip6_info.ip));
+    }
+}
+
+
+static void teardown_ppp(iface_info_t *info)
```
Comment (euripedesrocha) [inline examples/esp_netif/multiple_netifs/main/ppp_connect.c:76]: Nitpick: Functions should be called `ppp_init` and `ppp_destroy` to keep consistency with our components.

**Review by euripedesrocha — APPROVED** [review]
Just a small nitpick.

**Pull request merged**